node/activation_service.go

```diff
@@ -0,0 +1,712 @@
+			if err := configure(c, *configPath, &conf); err != nil {
+				return err
+			}
+
+			// NOTE(dshulyak) this needs to be max level so that child logger can can be current level or below.
+			// otherwise it will fail later when child logger will try to increase level.
+			encoder := zapcore.NewConsoleEncoder(zap.NewDevelopmentEncoderConfig())
+			if conf.LOGGING.Encoder == config.JSONLogEncoder {
+				encoder = zapcore.NewJSONEncoder(zap.NewDevelopmentEncoderConfig())
+			}
+			lg := log.NewWithLevel("node", zap.NewAtomicLevelAt(zap.DebugLevel), encoder, events.EventHook())
+
+			app := New(WithConfig(&conf), WithLog(lg))
+
+			// os.Interrupt for all systems, especially windows, syscall.SIGTERM is mainly for docker.
+			ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+			defer cancel()
+
+			types.SetLayersPerEpoch(app.Config.LayersPerEpoch)
+			// ensure all data folders exist
+			if err := os.MkdirAll(app.Config.DataDir(), 0o700); err != nil {
+				return fmt.Errorf("ensure folders exist: %w", err)
+			}
+
+			if err := app.Lock(); err != nil {
+				return fmt.Errorf("getting exclusive file lock: %w", err)
+			}
+			defer app.Unlock()
+
+			if err := app.Initialize(); err != nil {
+				return fmt.Errorf("initializing app: %w", err)
+			}
+
+			err := app.LoadIdentities()
+			switch {
+			case errors.Is(err, fs.ErrNotExist):
+				app.log.Info("Identity file not found. Creating new identity...")
+				if err := app.NewIdentity(); err != nil {
+					return fmt.Errorf("creating new identity: %w", err)
+				}
+			case err != nil:
+				return fmt.Errorf("loading identities: %w", err)
+			}
+
+			// Don't print usage on error from this point forward
+			c.SilenceUsage = true
+
+			// This blocks until the context is finished or until an error is produced
+			err = app.StartActivationService(ctx)
+			cleanupCtx, cleanupCancel := context.WithTimeout(
+				context.Background(),
+				30*time.Second,
+			)
+			defer cleanupCancel()
+			done := make(chan struct{}, 1)
+			// FIXME: per https://github.com/spacemeshos/go-spacemesh/issues/3830
+			go func() {
+				app.Cleanup(cleanupCtx)
+				close(done)
+			}()
+			select {
+			case <-done:
+			case <-cleanupCtx.Done():
+				app.log.Error("app failed to clean up in time")
+			}
+			return err
+		},
+	}
+
+	configPath = cmd.AddFlags(c.PersistentFlags(), &conf)
+
+	// versionCmd returns the current version of spacemesh.
+	versionCmd := &cobra.Command{
+		Use:   "version",
+		Short: "Show version info",
+		Run: func(c *cobra.Command, args []string) {
+			fmt.Print(cmd.Version)
+			fmt.Println()
+		},
+	}
+	c.AddCommand(versionCmd)
+
+	relayCmd := cobra.Command{
+		Use:          "relay",
+		Short:        "Run relay server",
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := configure(c, *configPath, &conf); err != nil {
+				return err
+			}
+			return runRelay(c.Context(), &conf)
+		},
+	}
+	c.AddCommand(&relayCmd)
```

Comment:
I think the smeshing-serivce binary doesn't need the relay cmd, right @ivan4th ?

Reply:
Removed. If @ivan4th will confirm that it's needed - it can be quickly/easily added.